### DISCUSSION TITLE
CRNN  长文本识别

### DISCUSSION THREAD

**Discussion by Hiwyl** [PaddlePaddle/PaddleOCR#13628] [Q&A]
使用crnn进行中英混合场景训练，设置shape是[32,320]，max_text_length 25 
1. 在验证集精度已经很高了 98%
2. predict_rec.py 使用 onnx 固定 shape 为 32，320 进行推理精度差很多 50%

推理时候很多数据的长宽比都比较极端，最大到40:1，请问这种场景有好的解决方案吗？训练直接增大imgW是不是最优方案？:pray:

**Comment by SWHL** [comment]
改为不固定shape试试。
这种我能想到的就是增加这种数据会好一些

**Reply by Hiwyl** [reply]
dynamic shape对部署不是很友好

**Comment by UserWangZz** [comment]
推理数据和验证集数据是不是差别比较大啊，建议使用推理数据微调模型

**Reply by Hiwyl** [reply]
看了下 eval 精度高是因为 CTCLabelEncode 只统计 小于 max_seqlen的数据吧
